### PR TITLE
feat(hooks): Add `useSessionStorage` hook by factoring out core implementation of `useLocalStorage`

### PR DESCRIPTION
**Pull request by mturley**
Renames the `useLocalStorage.ts` file to `useStorage.ts` which now exports both `useLocalStorage` and `useSessionStorage` hooks based on the same implementation. The only difference is the underlying storage API used and the lack of `StorageEvent`s when using `useSessionStorage`.

Adds a docs page for `useSessionStorage` with similar explanations, a few similar examples and a reference to the `useLocalStorage` docs page.

### CODE REVIEW

**Comment by ibolton336** [comment]
Thanks for all the work on this. & Bonus points for legible docs 👍

File: src/hooks/useStorage/useSessionStorage.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      <TextContent className={spacing.mbMd}>
+        <Text component="h4">Add item:</Text>
+      </TextContent>
+      <Form isWidthLimited isHorizontal className={spacing.mbMd}>
+        <ValidatedTextInput isRequired fieldId="name" field={addForm.fields.name} />
+        <ValidatedTextInput fieldId="description" field={addForm.fields.description} />
+        <FormGroup fieldId="submit">
+          <Button
+            id="submit"
+            onClick={() => {
```
Comment (ibolton336) [inline src/hooks/useStorage/useSessionStorage.stories.tsx:91]: Can we reset the form on "Add"? Seeing the value stick around after submitting.

**Review by ibolton336 — APPROVED** [review]
Reading through it everything makes sense. Just one subjective nit with the submit button on the complex value example.

File: src/hooks/useStorage/useSessionStorage.stories.tsx (hunk 1 above)
Comment (mturley) [reply src/hooks/useStorage/useSessionStorage.stories.tsx:91]: Oh yeah, not sure how I missed that in both `useLocalStorage` and `useSessionStorage` docs. Thanks!

Comment (mturley) [reply src/hooks/useStorage/useSessionStorage.stories.tsx:91]: Done

**Comment by ibolton336** [comment]
LGTM!

**Review by ibolton336 — APPROVED** [review]
LGTM!

**Pull request merged**